sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -157,6 +145,26 @@ object TypeCoercion {
     })
   }
 
+  /**
+   * Similar to [[findWiderCommonType]], but can't promote to string.
+   */
+  private def findWiderTypeWithoutStringPromotion(types: Seq[DataType]): Option[DataType] = {
```

Comment:
It is weird that its name is `findWiderTypeWithoutStringPromotion` because `findTightestCommonTypeOfTwo` is used inside. Also, let's add more docs to this method.